tests/sentry/api/test_event_search.py

```diff
@@ -964,6 +964,70 @@ def test_message_negative(self):
             ]]
         }
 
+    def test_boolean_term_simple(self):
+        assert get_snuba_query_args('user.email:foo@example.com AND user.email:bar@example.com') == {
+            'conditions': [
+                ['and', [
+                    ['email', '=', 'foo@example.com'],
+                    ['email', '=', 'bar@example.com']
+                ]]
+            ],
+            'filter_keys': {},
+            'has_boolean_terms': True,
+        }
+        assert get_snuba_query_args('user.email:foo@example.com OR user.email:bar@example.com') == {
+            'conditions': [
+                ['or', [
+                    ['email', '=', 'foo@example.com'],
+                    ['email', '=', 'bar@example.com']
+                ]]
+            ],
+            'filter_keys': {},
+            'has_boolean_terms': True,
+        }
+        assert get_snuba_query_args(
+            'user.email:foo@example.com AND user.email:bar@example.com OR user.email:foobar@example.com AND user.email:hello@example.com AND user.email:hi@example.com OR user.email:foo@example.com AND user.email:bar@example.com OR user.email:foobar@example.com AND user.email:hello@example.com AND user.email:hi@example.com'
+        ) == {
+            'conditions': [
+                ['or', [
+                    ['and', [
+                        ['email', '=', 'foo@example.com'],
+                        ['email', '=', 'bar@example.com']
+                    ]],
+                    ['or', [
+                        ['and', [
+                            ['email', '=', 'foobar@example.com'],
+                            ['and', [
+                                ['email', '=', 'hello@example.com'],
+                                ['email', '=', 'hi@example.com']
+                            ]]
+                        ]],
+                        ['or', [
+                            ['and', [
+                                ['email', '=', 'foo@example.com'],
+                                ['email', '=', 'bar@example.com']
+                            ]],
+                            ['and', [
+                                ['email', '=', 'foobar@example.com'],
+                                ['and', [
+                                    ['email', '=', 'hello@example.com'],
+                                    ['email', '=', 'hi@example.com']
+                                ]]
+                            ]]
+                        ]]
+                    ]]
+                ]]
+            ],
+            'filter_keys': {},
+            'has_boolean_terms': True,
+        }
+
+    def test_boolean_special_cases(self):
```

Comment:
@wedamija do we want to be able to say `'project_id:1 OR project_id:2'` or `'start:time1 AND end:time2'`? Currently I'm not sure that works because I'm only appending to conditions in snuba... It might but I'm not sure. Why were these treated differently for search filters?

Reply:
We handle project ids separately via `filter_keys`, so they end up getting translated to an `IN` statement at the moment. Same for issue ids. I'm not sure what you mean about start and end?

Reply:
@wedamija the more I think about this the more I realize this is probably just fine as is. And you'll probably want to actually be able to `OR` different things together i.e. Issues. I think we should just leave it as is rather than banning inputing the fields that are treated differently for conditions